libcloudforensics/providers/azure/internal/account.py

```diff
@@ -217,3 +258,187 @@ def CreateDiskFromSnapshot(
                           disk.name,
                           disk.location,
                           disk.zones)
+
+  def CreateDiskFromSnapshotURI(
+      self,
+      snapshot: compute.AZSnapshot,
+      snapshot_uri: str,
+      region: Optional[str] = None,
+      disk_name: Optional[str] = None,
+      disk_name_prefix: Optional[str] = None,
+      disk_type: str = 'Standard_LRS') -> compute.AZDisk:
+    """Create a new disk based on a SAS snapshot URI.
+
+    This is useful if e.g. one wants to make a copy of a disk in a separate
+    Azure account. This method will create a temporary Azure Storage account
+    within the destination account, import the snapshot from a downloadable
+    link (the source account needs to share the snapshot through a SAS link)
+    and then create a disk from the VHD file saved in storage. The Azure
+    storage account is then deleted.
+
+    Args:
+      snapshot (AZSnapshot): Source snapshot to use.
+      snapshot_uri (str): The URI of the snapshot to copy.
+      region (str): Optional. The region in which to create the disk. If not
+          provided, the disk will be created in the default_region associated to
+          the AZAccount object.
+      disk_name (str): Optional. String to use as new disk name.
+      disk_name_prefix (str): Optional. String to prefix the disk name with.
+      disk_type (str): Optional. The sku name for the disk to create. Can be
+          Standard_LRS, Premium_LRS, StandardSSD_LRS, or UltraSSD_LRS.
+          Default is Standard_LRS.
+
+    Returns:
+      AZDisk: Azure Compute Disk.
+
+    Raises:
+      RuntimeError: If the disk could not be created.
+    """
+
+    if not region:
+      region = self.default_region
+
+    # Create a temporary Azure account storage to import the snapshot
+    storage_account_name = hashlib.sha1(
```

Comment:
Why use `sha1` instead of a random hex blob?

Reply:
No particular reason, just wanted something that was linked to the particular snapshot (but there again, no requirement to have that). Let me know if you want this changed to a random hex blob.